sonic-stormond/scripts/stormond

```diff
@@ -0,0 +1,384 @@
+                dynamic_kvp_dict["health"] = storage_object.get_health()
+                dynamic_kvp_dict["temperature"] = storage_object.get_temperature()
+                dynamic_kvp_dict["latest_fsio_reads"] = storage_object.get_fs_io_reads()
+                dynamic_kvp_dict["latest_fsio_writes"] = storage_object.get_fs_io_writes()
+                dynamic_kvp_dict["disk_io_reads"] = storage_object.get_disk_io_reads()
+                dynamic_kvp_dict["disk_io_writes"] = storage_object.get_disk_io_writes()
+                dynamic_kvp_dict["reserved_blocks"] = storage_object.get_reserved_blocks()
+
+                dynamic_kvp_dict["total_fsio_reads"], dynamic_kvp_dict["total_fsio_writes"] = self._reconcile_fsio_rw_values(dynamic_kvp_dict, storage_device)
+
+                self.log_info("Storage Device: {}, Firmware: {}, health: {}%, Temp: {}C, FS IO Reads: {}, FS IO Writes: {}".format(\
+                storage_device, dynamic_kvp_dict["firmware"], dynamic_kvp_dict["health"], dynamic_kvp_dict["temperature"], dynamic_kvp_dict["total_fsio_reads"],dynamic_kvp_dict["total_fsio_writes"]))
+                self.log_info("Latest FSIO Reads: {}, Latest FSIO Writes: {}".format(dynamic_kvp_dict["latest_fsio_reads"], dynamic_kvp_dict["latest_fsio_writes"]))
+                self.log_info("Disk IO Reads: {}, Disk IO Writes: {}, Reserved Blocks: {}".format(dynamic_kvp_dict["disk_io_reads"], dynamic_kvp_dict["disk_io_writes"], \
+                dynamic_kvp_dict["reserved_blocks"]))
+
+                # update Storage Device Status to DB
+                self.update_storage_info_status_db(storage_device, self.dynamic_fields, dynamic_kvp_dict)
+
+            except Exception as ex:
+                self.log_info("get_dynamic_fields() failed with: {}".format(str(ex)))
+
+    
+   # Override signal handler from DaemonBase
+    def signal_handler(self, sig, frame):
+        FATAL_SIGNALS = [signal.SIGINT, signal.SIGTERM]
+        NONFATAL_SIGNALS = [signal.SIGHUP]
+
+        global exit_code
+
+        if sig in FATAL_SIGNALS:
+            self.log_info("Caught signal '{}'".format(SIGNALS_TO_NAMES_DICT[sig]))
+            
+            self.log_info("Syncing latest procfs reads and writes to disk")
+            self._sync_fsio_rw_json()
+
+            self.log_info("Exiting with {}".format(SIGNALS_TO_NAMES_DICT[sig]))
+
+            # Make sure we exit with a non-zero code so that supervisor will try to restart us
+            exit_code = 128 + sig  
+            self.stop_event.set()
+        elif sig in NONFATAL_SIGNALS:
+            self.log_info("Caught signal '{}' - ignoring...".format(SIGNALS_TO_NAMES_DICT[sig]))
+        else:
+            self.log_warning("Caught unhandled signal '{}' - ignoring...".format(SIGNALS_TO_NAMES_DICT[sig]))
+
+    # Main daemon logic
+    def run(self):
+
+        # Connect to CONFIG_DB and get polling and sync intervals --
+        # this is to be able to dynamically configure the polling and sync times.
```

Comment:
@assrinivasan HLD says it reads only once and need to restart the daemon.?

Reply:
HLD says:

Shortly succeeding the initialization process, the daemon will query the Config DB for fields called daemon_polling_interval and fsstats_sync_interval within a newly proposed table STORMOND_CONFIG|INTERVALS and use the value to set the looping frequency for getting dynamic informaton and syncing this information to disk, respectively. In the absense of this table/field, we would default to 3600 seconds for the polling interval and 86400 seconds for sync interval.

NOTE: **Modifying the configuration intervals will only be applied in the subsequent cycle. Should immediate implementation be desired by the user, a configuration reload and daemon restart would be necessary.**